libraries/chain/platform_timer_asio_fallback.cpp

```diff
@@ -75,18 +75,25 @@ void platform_timer::start(fc::time_point tp) {
 }
 
 void platform_timer::expire_now() {
-   bool expected = false;
-   if (expired.compare_exchange_strong(expected, true)) {
+   state_t expected = state_t::running;
+   if (_state.compare_exchange_strong(expected, state_t::timed_out)) {
+      call_expiration_callback();
+   }
+}
+
+void platform_timer::interrupt_timer() {
+   state_t expected = state_t::running;
+   if (_state.compare_exchange_strong(expected, state_t::interrupted)) {
```

Comment:
Why don't these call `my->timer->cancel()`?

Isn't it possible that someone calls `interrupt_timer() ` (which updates the state but doesn't cancel the `async_wait`), and then calls `start(fc::time_point::maximum())`, and may be surprised when the previous `async_wait` expires the timer.

Reply:
We would need to add a mutex around the timer to do that. I think the intention is that stop() should be called before calling start() again.